homeassistant/components/melcloud/climate.py

```diff
@@ -162,6 +162,7 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
                     ATTR_VANE_HORIZONTAL_POSITIONS: self._device.vane_horizontal_positions,
                 }
             )
+            self._attr_supported_features |= ClimateEntityFeature.SWING_HORIZONTAL_MODE
```

Comment:
We should not set this during `extra_state_attributes` as that is a side effect. If for example extra_state_attributes wouldn't be called for what reason, the supported feature is not added, and that is bad.

Reply:
Before submitting this PR I tried checking for self._device.vane_horizontal (to check if the device supports horizontal vanes) in __init__ but that didn't work since the device isn't initialized until the device information is obtained from the cloud.

So I tried reimplementing the MelCloudClimate.async_update in AtaDeviceClimate and check for horizontal vane support when the MelCloudClimate.async_update returns. I tested that works fine here too, so I submitted that change in a new commit.

Reply:
I've marked this as resolved since I submitted a fix for the issue you commented. If you think it's still wrong, please reopen it.